transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -148,7 +148,7 @@ internal class MessagingClientImpl(
         checkConfigured()
         if (messageStore.startOfConversation) {
             log.i { "All history have been fetched." }
-            return
+            throw UnsupportedOperationException("All history have been fetched.")
```

Comment:
I'm not a fan of throwing in cases like this--It's not really an exceptional situation, is it?
(BTW If you do decide to throw, the language should be "All history has been fetched.")

Reply:
Other options were: 
Add boolean return type to the function with indicator of success/fail. But this has a downside of changing public api. Also it sounds a little bit unapropriate.
Second option would be simply allow network request go through and this will result in empty history response, but this also sounds wrong, as I wanted to avoid unnecessary network requests.

Reply:
Does adding an exception here not change the public api?

Reply:
No. This function is already defined as throwable, therefore anybody who use it aware that exception can be thrown.

Reply:
Ah, ok. That's fair.

Reply:
I like this very much.

Reply:
I like this too.